Source/EasyNetQ/Consumer/HandlerRunner.cs

```diff
@@ -141,8 +141,17 @@ private string BuildErrorMessage(ConsumerExecutionContext context, Exception exc
 
         public void Dispose()
         {
-            consumerErrorStrategy.Dispose();
+            Dispose(true);
+            GC.SuppressFinalize(this);
```

Comment:
Looks like it's still best practice to call `GC.SuppressFinalize(this)` as the can be inherited and the subclass theoretically could have a finalizer, so would you mind if I leave it like this? http://stackoverflow.com/questions/18336856/implementing-idisposable-correctly